editoast/src/views/timetable/stdcm.rs

```diff
@@ -543,21 +608,17 @@ fn build_single_margin(margin: Option<MarginValue>) -> Margins {
 }
 
 /// Build the list of work schedules for the given time range
-async fn build_work_schedules(
-    conn: &mut DbConnection,
-    time: DateTime<Utc>,
+fn build_stdcm_work_schedules(
```

Comment:
`build_stdcm_work_schedule` and `work_conflict_work_schedule` are very similar. Shouldn't they share some code? I don't understand why we do have 2 different functions that almost do the same thing... but not quite. Available to discuss it if you need.

Reply:
```suggestion
fn filter_work_schedules_time_window(
```

The WorkSchedule list is now provided in argument. Agreed about factorizing the filtering part.

Reply:
Done.